mbos

aliyailina opened the pull request:
Rewrite LeftTabbedPageRenderer.apple.cs for imitating lazy initialization

Commit:
Rewrite LeftTabbedPageRenderer.apple.cs for imitate lazy initialization

angpysha commented on Angpysha.LeftTabbedPage/iOS/LeftTabbedPageRenderer.apple.cs:
```diff
@@ -1,598 +1,317 @@
+            if (!_queuedSize.IsZero)
             {
-                TabBar.LeftTabbedPageWeak = new WeakReference<Shared.LeftTabbedPage>(leftTabbedPage);
+                Element.Layout(new Rectangle(Element.X, Element.Y, _queuedSize.Width, _queuedSize.Height));
```
Width should be calculated using this formula _queuedSize.With-_tabbarWidth

aliyailina replied:
No, it shouldn't, because Element - is the whole page with area below tab bar too (you can check the renderer and see, that tab bar is adding to View in which TabPage renders; also you can check the default Xamarin.iOS TabbedRenderer to certain that it shouldn't (https://github.com/xamarin/Xamarin.Forms/blob/80a4dbfb1cd897f478c38d0aafa6a5d67940d2f3/Xamarin.Forms.Platform.iOS/Renderers/TabbedRenderer.cs, line:145 nothing is subtracted from _queueSize.Height, size fits using ContainerArea property). 

PS. update to fit any page in renderer, not only TabPage was commited

angpysha commented on Angpysha.LeftTabbedPage/iOS/LeftTabbedPageRenderer.apple.cs:
```diff
@@ -1,598 +1,317 @@
         }
 
-        void OnPagePropertyChanged(object sender, PropertyChangedEventArgs e)
+        public SizeRequest GetDesiredSize(double widthConstraint, double heightConstraint)
```
Where it is using?

aliyailina replied:
GetDesiredSize(double widthConstraint, double heightConstraint) is the method of IVisualElementRenderer interface

Commit:
Remove necessity to use special pages inside tabbed page

LeftTabbedPageRenderer.apple.cs:
remove useless vars;
fix initializing index issue;
add functionality to automatically fit inner page to tabbed
